### PR TITLE
[irods/irods#7098] iadmin: Add pam_password config subcommands (main)

### PR DESCRIPTION
**Pull request by alanking**
In service of irods/irods#7098

Companion PR: https://github.com/irods/irods/pull/7245

Will take out of draft when https://github.com/irods/irods/pull/7245 is out of draft.

### CODE REVIEW

File: src/iadmin.cpp
Comment (alanking) [inline src/iadmin.cpp:2805]: Not sure why this is here twice. Will look into it.

I also think I forgot to add my help messages in all the right places.

File: src/iadmin.cpp
Comment (alanking) [inline src/iadmin.cpp:2695]: clang-format betrayed me

File: src/iadmin.cpp
Comment (alanking) [inline src/iadmin.cpp:2669]: ...and name them better

File: src/iadmin.cpp
Comment (alanking) [inline src/iadmin.cpp:245]: Maybe use keywords for these

File: src/iadmin.cpp
Comment (trel) [inline src/iadmin.cpp:262]: 2700 hard coded here...  probably never going to change, but could be 'derived', same as the check in the line above...

Comment (alanking) [reply src/iadmin.cpp:262]: Yep, will do

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:2805]: "help" and "h" are recognized by `iadmin`, which is why there are two of these.

Comment (trel) [reply src/iadmin.cpp:2805]: well, feels weird, but explained!

File: src/iadmin.cpp
Comment (korydraughn) [reply src/iadmin.cpp:2695]: Lol, disable clang-format for that part.

File: src/iadmin.cpp
Comment (korydraughn) [inline src/iadmin.cpp:189]: Consider renaming so that usage reads like a question.

- `is_connected_as_rodsadmin`
- `is_rodadmin`
- `is_privileged_user`
- `connected_user_is_rodsadmin`

Those are just some examples. Feel free to come up with something else.

File: src/iadmin.cpp
Comment (korydraughn) [inline src/iadmin.cpp:152]: `RcComm&`?

File: src/iadmin.cpp
Comment (korydraughn) [inline src/iadmin.cpp:238]: Consider checking the options first to avoid network traffic.

File: src/iadmin.cpp
Comment (korydraughn) [inline src/iadmin.cpp:2674]: Should we sort these? Hmm ...

File: src/iadmin.cpp
Comment (korydraughn) [inline src/iadmin.cpp:2683]: Why parentheses here?
Wouldn't this result in the line being really long in the terminal?

Comment (alanking) [reply src/iadmin.cpp:2683]: Yep. The compiler gives a warning here because if you split up the string it isn't sure if you meant to put a comma at the end of the line. I'm happy to just split this into two lines in the output if that's preferable.

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:189]: The idea behind the name was to hint at side effects (the function will print messages for various errors that occur) and not give the impression that it's a pure function. I see that it did not do its job well :p 

If we're okay with the side effects, I'll just rename it to `connected_user_is_rodsadmin` which is what I had originally. Otherwise, I'll think about a better approach here (to me, top candidate is just not using a function).

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:152]: Sure

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:238]: Will do

File: src/iadmin.cpp
Comment (korydraughn) [reply src/iadmin.cpp:2683]: Don't we want these lines to be split via a comma so the help text honors the imaginary margin shared by all lines?

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:189]: The side effect of the function which is only supposed to return whether the connected user is a rodsadmin is that it also prints messages for various errors that occur.

At least in my mind, a "clean" implementation of this function would be to only return whether the connected user is a rodsadmin and leave it up to the caller to print errors and whatnot. That's all I was getting at there. Might not be a big deal that the function prints things - I mostly did this because I didn't want to see all that mess in `print_pam_password_config`

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:2674]: Seeing as how we will be updating the docs anyway, we can sort them in all the places

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:2695]: Actually I think this is right. It at least looks like the `get_delay_server_info` help text, so I'm just going to leave it.

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp:189]: I've decided to just remove this function. Some code duplication now, but at least there's no hidden surprises :)

File: src/iadmin.cpp
Comment (trel) [reply src/iadmin.cpp:2674]: let's sort.   it's the most defensible position.

**Comment by alanking** [comment]
irods/irods#7245 has changed purpose. This will be updated soon to be in service of irods/irods#7274

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: This function has been removed. Resolving.

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: This function has been removed. Resolving.

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: Done

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: Done

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: Done, I think

**Comment by korydraughn** [comment]
What are next steps for this?

**Comment by alanking** [comment]
Preparing companion PR in irods/irods... Will notify when this is ready for review.

**Comment by alanking** [comment]
Ready for review

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: Done

File: src/iadmin.cpp
Comment (alanking) [reply src/iadmin.cpp]: Done

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Squash, pound, and merge when ready.

**Comment by alanking** [comment]
I'm gonna hold off and merge this at the same time as irods/irods#7331

**Pull request merged**
